### DISCUSSION TITLE
Docker error when trying to build own containers

### DISCUSSION THREAD

**Discussion by jonprattC78** [Azure/ipam#53] [Q&A]
Hi!

I have re-written the Bicep code to Terraform (in line with my company IaC standards. Would be happy to share the TF code if you are interested?)
So I have everything deployed - and when I come to trying to create the ipam-engine container to push to ACR - I follow the guide in the 'contributing' section and get an error:

----------------------------------------
docker build --rm --no-cache -t ipam-engine -f Dockerfile.prod .                                                                                                
[+] Building 0.0s (1/2)                                                                                                                                                    
 => [internal] load build definition from Dockerfile.prod                                                                                                             0.0s
 => => transferring dockerfile: 2B                                                                                                                                    0.0s
failed to solve with frontend dockerfile.v0: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount3841449649/Dockerfile.prod: no such file or directory

----------------------------------------

A quick google suggests it might be an issue with the build kit (but it was Stackoverflow so take that with a pinch of salt :) 

But I wondered if anyone else has seen issues when trying to build their containers. 
I am using an M1 Macbook, which also might be a contributing factor (not 100% sure right now) - and so I will try to build on a Win11 machine and see if I get the same failure.

**Comment by jonprattC78** [comment]
*update* the same error occurs on Win11 so its a Docker issue not a Apple Silicon issue

**Comment by jonprattC78** [comment]
Looks like I might have jumped ahead here, trying to build a .prod - as looking at the files, the prod docker file is missing / incomplete. 

the .dev container built fine, so I guess (looking at the files related to prod) that more customisation is required. 
As I can continue to test my Terraform deployed IPAM infra with a working .dev container then, for now, I am unblocked. 

But I would like to talk to you more on this subject another time - if that is ok?

**Reply by DCMattyG** [reply]
Hi @jonprattC78, please reach out to me 1:1 via email to Matthew.Garrett@microsoft.com. I'm happy to spend as much time as you need going through this with you and helping you find a solution.
